tests/unit/test_dimension_utilities.py

```diff
@@ -304,6 +312,141 @@ def test_prefetch_dimension_variables(self, mock_get_opendap_nc4):
+        prefetch_dataset_name = 'tests/data/ATL16_prefetch.nc4'
+        varinfo_prefetch = VarInfoFromDmr(
+                'tests/data/ATL16_prefetch.dmr'
+        )
+        required_dimensions = {'/npolar_grid_lat','/npolar_grid_lon',
+                               '/spolar_grid_lat','/spolar_grid_lon',
+                               '/global_grid_lat','/global_grid_lon'}
+
+        with self.subTest('Bounds need to be written'):
+            mock_needs_bounds.return_value = True
+            add_bounds_variables(prefetch_dataset_name,
+                                 required_dimensions,
+                                 varinfo_prefetch,
+                                 self.logger)
+            self.assertEqual(mock_write_bounds.call_count, 6)
+
+            mock_needs_bounds.reset_mock()
+            mock_write_bounds.reset_mock()
+
+        with self.subTest('Bounds should not be written'):
+            mock_needs_bounds.return_value = False
+            add_bounds_variables(prefetch_dataset_name,
+                                 required_dimensions,
+                                 varinfo_prefetch,
+                                 self.logger)
+            mock_write_bounds.assert_not_called()
+
+
+    def test_needs_bounds(self):
+        """ Ensure that the correct boolean value is returned for four
+            different cases:
+
+            1) False - cell_alignment[edge] attribute exists and bounds variable already exists.
+            2) False - cell_alignment[edge] attribute does not exist and bounds variable already exists.
+            3) True - cell_alignment[edge] attribute exists and bounds variable does not exist.
+            4) False - cell_alignment[edge] attribute does not exist and bounds variable does not exist.
+
+        """
+        varinfo_bounds = VarInfoFromDmr(
+            'tests/data/ATL16_prefetch_bnds.dmr'
+        )
+
+        with self.subTest('Variable has cell alignment and bounds'):
+            self.assertFalse(needs_bounds(varinfo_bounds.get_variable('/variable_edge_has_bnds')))
+
+        with self.subTest('Variable has no cell alignment and has bounds'):
+            self.assertFalse(needs_bounds(varinfo_bounds.get_variable('/variable_no_edge_has_bnds')))
+
+        with self.subTest('Variable has cell alignment and no bounds'):
+            self.assertTrue(needs_bounds(varinfo_bounds.get_variable('/variable_edge_no_bnds')))
+
+        with self.subTest('Variable has no cell alignment and no bounds'):
+            self.assertFalse(needs_bounds(varinfo_bounds.get_variable('/variable_no_edge_no_bnds')))
+
+
+    def test_get_bounds_array(self):
+        """ Ensure that the expected bounds array is created given
+            the input dimension variable values.
+
+        """
+        prefetch_dataset = Dataset('tests/data/ATL16_prefetch.nc4', 'r')
+        dimension_path = '/npolar_grid_lat'
+
+        bounds_array = np.array([[90.0, 89.0], [89.0, 88.0], [88.0, 87.0], [87.0, 86.0],
```

Comment:
```suggestion
        expected_bounds_array = np.array([[90.0, 89.0], [89.0, 88.0], [88.0, 87.0], [87.0, 86.0],
```
and below where you use it